### ISSUE TITLE
Add Debug Logging to SchluterAPI Class for Improved Error Diagnosis

### ISSUE THREAD

**Issue by winterborn** [derekprior/homebridge-schluter-thermostat#26]
@winterborn though things to be working fine, the plugin is throwing a lot of exceptions.
[homebridge-schluter-thermostat.log.txt](https://github.com/user-attachments/files/17451725/homebridge-schluter-thermostat.log.txt)

_Originally posted by @betabob in https://github.com/derekprior/homebridge-schluter-thermostat/issues/16#issuecomment-2425178283_

Added debug logs to assist resolving intermittent issues where API responses are missing expected data fields, resulting in TypeError exceptions (e.g., "Cannot read properties of undefined").

Affected Methods:

```
thermostatState
getTemperatureUnit
signIn
```

**Comment by winterborn** [comment]
@derekprior @betabob, adding extra debug logs here to check responses to determine issue.

if @betabob could either pm me or provide additional logs with above PR, it should give me a better idea of why it keeps throwing errors...

**Comment by betabob** [comment]
@winterborn how to pm you? I don't believe Github supports direct messaging between users, does it?

**Comment by winterborn** [comment]
> @winterborn how to pm you? I don't believe Github supports direct messaging between users, does it?

You are right, no worries, feel free to use my email address: phwhite89@gmail.com or paste the logs here.

**Comment by betabob** [comment]
@winterborn how can I test your PR if it hasn't been published with a new version of the plugin yet?

**Comment by 113602611780178409309** [comment]
> [@derekprior](https://github.com/derekprior) [@betabob](https://github.com/betabob)เพิ่มบันทึกการดีบักเพิ่มเติมที่นี่เพื่อตรวจสอบการตอบสนองเพื่อระบุปัญหา
> 
> ถ้า[@betabob](https://github.com/betabob)คุณสามารถส่งข้อความส่วนตัวถึงฉันหรือส่งบันทึกเพิ่มเติมพร้อม PR ข้างต้นให้ฉันได้ ซึ่งน่าจะทำให้ฉันเข้าใจได้ดีขึ้นว่าทำไมถึงเกิดข้อผิดพลาดขึ้นอยู่เรื่อยๆ...

**Comment by 113602611780178409309** [comment]
https://github.com/derekprior/homebridge-schluter-thermostat/issues/26
